### PR TITLE
Various updates/nits based on DX improvements

### PR DESCRIPTION
**Pull request by carsonfarmer**
## Description

Here are a set of changes to the DX flow of using Database, as 'discovered' by working through examples and highlighting pain points. The following issues reflect the fixes and updates that this PR addresses.

Fixes...

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## How Has This Been Tested?

All existing tests should continue to pass, and any new APIs/methods have corresponding tests added.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

> This PR template comes from https://github.com/embeddedartistry/templates

### CODE REVIEW

File: packages/database/src/db.spec.ts [hunk 1]
```diff
@@ -56,7 +56,7 @@ async function runListenersComplexUseCase(los: string[]) {
   const events: Update[] = []
   // @todo: Should we wrap this in a 'listen' method instead?
   for (const name of los) {
-    db.on(name, (update: any) => {
+    db.emitter.on(name, (update: any) => {
```
Comment (carsonfarmer) [inline packages/database/src/db.spec.ts:59]: This is a 'breaking' change here @andrewxhill

File: packages/database/src/db.spec.ts [hunk 2]
```diff
@@ -95,7 +95,7 @@ async function runListenersComplexUseCase(los: string[]) {
   // Collection2 delete i2
   await Collection1.delete(i2._id)
 
-  db.removeAllListeners()
+  db.emitter.removeAllListeners()
```
Comment (carsonfarmer) [inline packages/database/src/db.spec.ts:98]: here also

File: packages/database/src/db.spec.ts [hunk 3]
```diff
@@ -155,17 +152,27 @@ describe('Database', () => {
...
       await dummy1.save()
 
-      await delay(2000)
+      await delay(1000)
       const dummy2 = await Dummy2.findById(dummy1._id)
       expect(dummy2.name).to.equal(dummy1.name)
       expect(dummy2.counter).to.equal(dummy1.counter)
+
+      // Now the other way around
+      dummy2.counter = 13
```
Comment (carsonfarmer) [inline packages/database/src/db.spec.ts:165]: this is a test to ensure that issue you found previously is now fixed (around missing log id)

File: packages/database/src/db.ts [hunk 4]
```diff
@@ -95,6 +95,16 @@ export class Database extends EventEmitter2 implements DatabaseSettings {
...
 
+  /**
+   * The event emitter that can be used by subscribers to follow database updates.
+   * Event names are structured as <collection>.<id>.<type>, and they support 'wildcard'
+   * matching, so `emitter.many(['foo', '*', Op.Type.Delete], callback)` will match all
+   * delete operations on the 'foo' collection. Similarly, `emitter.on('foo.**', callback)` will
+   * match all event types on the 'foo' collection. To observe a given instance, try
+   * `emitter.on('foo.${instance._id}', callback)`. See EventEmitter2 docs for further details.
+   */
+  public emitter: EventEmitter2 = new EventEmitter2({ wildcard: true })
```
Comment (carsonfarmer) [inline packages/database/src/db.ts:106]: moved the emitter to its own property, and added some docs

File: packages/database/src/db.ts [hunk 5]
```diff
@@ -312,7 +323,7 @@ export class Database extends EventEmitter2 implements DatabaseSettings {
       if (update.type !== undefined) {
         event.push(update.type.toString())
       }
-      this.emit(event, update)
+      this.emitter.emit(event, update)
```
Comment (carsonfarmer) [inline packages/database/src/db.ts:326]: also relevant here.

File: packages/database/src/db.spec.ts (hunk 1 above)
Comment (andrewxhill) [reply packages/database/src/db.spec.ts:59]: right, so in my new example, i'll need to do `fromEvent(db.emitter)` now instead of just db

Comment (carsonfarmer) [reply packages/database/src/db.spec.ts:59]: exactly

**Pull request merged**